zerver/models.py

```diff
@@ -1780,6 +1780,25 @@ def get_user_including_cross_realm(email: str, realm: Optional[Realm]=None) -> U
 def get_system_bot(email: str) -> UserProfile:
     return UserProfile.objects.select_related().get(email__iexact=email.strip())
 
+def get_cross_realm_bot_by_id(uid: int) -> Optional[UserProfile]:
+    for email in settings.CROSS_REALM_BOT_EMAILS:
+        bot = get_system_bot(email)
+        if bot.id == uid:
+            return bot
+
+    return None
+
+def get_user_by_id_in_realm_including_cross_realm(
+        uid: int,
+        realm: Optional[Realm]=None
```

Comment:
It's not clear to me why you're making `Realm` optional here.  Even though we're allowing cross-realm bots to be returned here, it seems like we still always know the relevant realm to search for the non-bot case.  In general optional parameters are to be avoided.  They're a necessary evil at times, but I don't you need this to be optional here.

Reply:
@showell: I'm just following the pattern followed by `get_user_including_cross_realm`, which is what we use currently. The reason that made sense to me was that if someone deliberately tries to look for a cross-realm bot, the realm is irrelevant, isn't it? Still, my primary objective was to replicate `get_user_including_cross_realm` closely, which might not have been a worthwhile goal, to begin with!